charts/linkerd-control-plane/templates/destination.yaml

```diff
@@ -208,6 +208,10 @@ spec:
         {{- range (.Values.destinationController).experimentalArgs }}
         - {{ . }}
         {{- end }}
+        {{- with (.Values.destinationController).experimentalEnv }}
+        env:
+        {{- toYaml . | nindent 8 -}}
```

Comment:
It looks like these  [configurations work differently from those of the proxy's experimentalEnv settings](https://github.com/linkerd/linkerd2/pull/11874/files) which are expressed as objects instead of as lists of name/values. Is this intentional'/

Reply:
It is intentional. It allows for things like this:
```yaml
destinationController:
  experimentalEnv:
  - name: ENV_VAR
    valueFrom:
      configMapKeyRef:
        key: env-var-key
        name: my-config-map
```